src/pathutil/vcs.go

```diff
@@ -7,38 +7,37 @@ package pathutil
 import (
 	"os"
 	"path/filepath"
+	"strings"
 )
 
-var vcsDirs = []string{".git", ".svn", ".hg"}
-
-// FindVCSRoot find package root path from arg path
-func FindVCSRoot(basedir string) string {
-	var foundVCSDir bool
-
-	findvcsDirWalkFunc := func(path string, info os.FileInfo, err error) error {
-		if err != nil || info == nil || info.IsDir() == false {
-			return nil
-		}
-
-		for _, d := range vcsDirs {
-			_, err := os.Stat(filepath.Join(path, d))
-			if err == nil {
-				foundVCSDir = true
-				break
-			}
-		}
-
-		return nil
+// FindVCSRoot finds the vcs root directory from path.
+func FindVCSRoot(path string) string {
+	var vcsDirs = []string{".git", ".svn", ".hg", "_darcs"}
+	if !IsDir(path) {
+		path = filepath.Dir(path)
 	}
 
+	var found bool
 	for {
-		filepath.Walk(basedir, findvcsDirWalkFunc)
-		if !foundVCSDir {
-			basedir = filepath.Dir(basedir)
-			continue
+		filepath.Walk(path, func(p string, info os.FileInfo, err error) error {
+			if err != nil || info == nil || info.IsDir() == false {
```

Comment:
**[gosimple]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
should omit comparison to bool constant, can be simplified to !info.IsDir() (S1002)